frappe/utils/data.py

```diff
@@ -2015,30 +2013,18 @@ def get_filter(doctype: str, f: dict | list | tuple, filters_config=None) -> "fr
 	from frappe.database.utils import NestedSetHierarchy
 	from frappe.model import child_table_fields, default_fields, optional_fields
 
-	if isinstance(f, dict):
-		key, value = next(iter(f.items()))
-		f = make_filter_tuple(doctype, key, value)
-
-	if not isinstance(f, list | tuple):
-		frappe.throw(frappe._("Filter must be a tuple or list (in a list)"))
-
-	if len(f) == 3:
-		f = (doctype, f[0], f[1], f[2])
-	elif len(f) > 4:
-		f = f[0:4]
```

Comment:
Here it is where 5th and forth following values silently dropped before, afaics, rendering the respective test data slack data that now fails the runtime the type checks.

Reply:
There seems to be a frontend usage hinted at [here](https://github.com/frappe/frappe/blob/e8808cec17d100dab42938d91d2ff2789dfe4192/frappe/public/js/frappe/list/base_list.js#L478-L480), but it's not passed to the server.